src/emulator/dataconnect/pgliteServer.ts

```diff
@@ -0,0 +1,119 @@
+// https://github.com/supabase-community/pg-gateway
+
+import { PGlite } from "@electric-sql/pglite";
+// Unfortunately, we need to dynamically import the Postgres extensions.
+// They are only available as ESM, and if we import them normally,
+// our tsconfig will convert them to requires, which will cause errors
+// during module resolution.
+const { dynamicImport } = require(true && "../../dynamicImport");
+import * as net from "node:net";
+import {
+  getMessages,
+  type PostgresConnection,
+  FrontendMessageCode,
+  BackendMessageCode,
+} from "./pg-gateway/index";
+import { fromNodeSocket } from "./pg-gateway/platforms/node";
+import { logger } from "../../logger";
+
+export class PostgresServer {
+  public callCount = 0;
+  private username: string;
+  private database: string;
+
+  public db: PGlite | undefined;
+  public async createPGServer(): Promise<net.Server> {
+    const db: PGlite = await this.getDb();
+    await db.waitReady;
+    const server = net.createServer(async (socket) => {
+      const connection: PostgresConnection = await fromNodeSocket(socket, {
+        serverVersion: "16.3 (PGlite 0.2.0)",
+        auth: { method: "trust" },
+
+        async onMessage(data: Uint8Array, { isAuthenticated }: { isAuthenticated: boolean }) {
+          // Only forward messages to PGlite after authentication
+          if (!isAuthenticated) {
+            return;
+          }
+          const result = await db.execProtocolRaw(data);
+          // Extended query patch removes the extra Ready for Query messages that
+          // pglite wrongly sends.
+          return extendedQueryPatch.filterResponse(data, result);
+        },
+      });
+
+      const extendedQueryPatch: PGliteExtendedQueryPatch = new PGliteExtendedQueryPatch(connection);
+
+      socket.on("end", () => {
+        logger.debug("Postgres client disconnected");
+      });
+    });
+    const listeningPromise = new Promise<void>((resolve) => {
+      server.listen(5432, "127.0.0.1", () => {
```

Comment:
Is this always the port, or is this something configurable?

Reply:
Ahh, good catch, we should make this configurable. I'll put it in firebase.json#emulators.dataconnect.postgresPort/Host and use these as the defaults